### PR TITLE
refactor: use es6 classes

### PR DESCRIPTION
**Pull request by andreasgan**
Adds babel and uses es6 classes instead of "class.extend"

### CODE REVIEW

**Comment by jthomerson** [comment]
@yokuze whether we need Babel or not is a question for you.

**Comment by andreasgan** [comment]
Even if babel doesn't add much value in this specific case, it makes development easier when you can use new features without having to think too much about browser support

File: src/js/chromecast/ChromecastSessionManager.js [hunk 1]
```diff
@@ -189,56 +183,58 @@ ChromecastSessionManager = Class.extend(/** @lends ChromecastSessionManager.prot
...
 
    /**
     * @see https://developers.google.com/cast/docs/reference/chrome/cast.framework.CastContext
     * @returns {object} the current CastContext, if one exists
     */
-   getCastContext: getCastContext,
+
+   getCastContext() {
+      return cast.framework.CastContext.getInstance();
+   }
```
Comment (kontrollanten) [inline src/js/chromecast/ChromecastSessionManager.js:195]: What about doing this to a static function? Then we can use that in `isChromecastConnected`.

**Review by kontrollanten — COMMENTED** [review]
It seems that uglify doesn't support ES6 syntax, so babel is needed to be able to switch to ES6 class.

**Comment by yokuze** [comment]
Thank you for the contribution @andreasgan. Closing this since https://github.com/silvermine/videojs-chromecast/pull/98/ is basically identical and was more recently active. 

FWIW, I agree that Babel is necessary. While the ES6 `class` keyword has good browser support, _ES2022_ `static` properties and methods are not supported in iOS < 14.1, which includes versions of iOS that we target: https://caniuse.com/?search=static%20class

Without Babel, it's easy to let new syntax and built-in methods slip in without knowing that they are incompatible with targeted environments. Babel's transpilation and automatic polyfilling reduces of the overhead for contributors and maintainers who would otherwise have to consult the browser support matrix for each non-ES6 syntax and built-in method that they intend to use.

**Pull request closed without merging**